tools/autograd/gen_variable_type.py

```diff
@@ -1352,13 +1385,51 @@ def emit_call(f: NativeFunction, unpacked_bindings: List[Binding]) -> str:
         else:
             guard = "at::AutoDispatchBelowADInplaceOrView guard;"
 
+        try_jit_decomposition_bool = "true" if try_jit_decomposition else "false"
+        any_has_forward_grad = (
```

Comment:
Why re-do the full condition calculation instead of using the boolean flag `_any_has_forward_grad_result` ?

Reply:
We cannot use that flag because that flag is only computed if there exists a formula for that output, i.e. it looks at the formula for `result` and checks what input's tangents it uses in its computation.